schema_validator.go

```diff
@@ -29,3 +32,74 @@ func (h *HostNameValidator) Validate(v interface{}) (munged interface{}, err err
 	}
 	return munged, jsonmap.NewValidationError("invalid type")
 }
+
+// PortValidator implements the jsonmap Validator interface.
+type PortValidator struct{}
+
+// Validate ensures that the port given is valid (empty OR non-zero uint16).
+// 0 ports are legal but will fail this validator.
+// v: the the value to be verified, should be a string
+// returns:
+//   munged: the, possibly modified, value
+//   err: any errors encountered
+func (h *PortValidator) Validate(v interface{}) (munged interface{}, err error) {
```

Comment:
I know you didn't establish this pattern, but I'd prefer to avoid using named return values.